ci-cd/github-actions-with-vercel-cicd/.github/workflows/e2e-release-production.yaml

```diff
@@ -0,0 +1,17 @@
+name: E2E-Release-Production
```

Comment:
The purpose of this comment is simply to provide information for discussion; no action is required.

Would it be possible to create a second file for feature branches only so that it will be clear that we want separate flow for preview and production? 

We still want to run E2E using QA data on PR before merging to production and then we will run E2E using staging or pre-prod/prod data

So the preview yaml file would have something like

```

on:
  push:
    branches-ignore:
    - master
    
```

Reply:
Please explain it better, by QA do you mean a normal push that creates a preview deployment? and why do we need staging and how would you do staging?

Reply:
A second yaml file can be created that runs e2e testing on pre-production environments should a team need to. Below is an example that would run e2e tests in pre-production and does not use `vercel alias`. A merge for a PR will be blocked if an e2e tests(s) fails with this action.

```yaml
name: E2E-Pre-Production
on: [deployment_status]
jobs:
  e2e:
    if: github.event.deployment.environment != 'Production' && github.event_name == 'deployment_status' && github.event.deployment_status.state == 'success'
    runs-on: ubuntu-latest
    steps:
      - name: Placeholder for running e2e tests via Github Actions (Cypress, Playwright, etc)
        run: [e2e command here]
```

This new example is focused on providing Vercel CI/D with e2e _and_ releasing to production (via `alias`), which is why I've only included an action for production. I can add another for a separate environment if that fills in any gaps we think should be addressed here. 

Lastly, I'll note that it is not possible to check whether a certain branch is deployed on `deployment_status` because the `deployment_status` event is triggered when a deployment status is updated, and not when the deployment is created. The `ref` field typically refers to the branch or tag that is being deployed, and since deployment statuses are not associated with a specific branch or tag, the `ref` field is not included in the `deployment_status` event payload. 
However, `github.event.deployment.environment` is available and will be set to Production, Preview or Development, which is leveraged in this example.

https://docs.github.com/en/actions/reference/events-that-trigger-workflows#deployment-status
https://docs.github.com/en/actions/reference/events-that-trigger-workflows#deployment